### DISCUSSION TITLE
Unknown language: "B4X"

### DISCUSSION THREAD

**Discussion by pyhoon** [github-linguist/linguist#7033] [Q&A] [answered]
Hi, may I know why when I click on the link under Languages https://github.com/pyhoon/B4A-Bridge/search?l=b4x
I am redirecting to a search result with 0 match.
https://github.com/search?q=repo%3Apyhoon%2FB4A-Bridge++language%3AB4X&type=code

<img width="839" alt="unknown-language-b4x" src="https://github.com/user-attachments/assets/7f69e0c5-176a-47d0-8288-f3b938871236">

**Comment by lildude** [comment, marked as answer]
This is a known issue due to the fact GitHub's Search and Linguist are not tightly linked... Search lags behind Linguist.

Support for B4X was added in [Linguist v8.0.0](https://github.com/github-linguist/linguist/releases) released on Monday and this is why I always add this not to the release PRs like https://github.com/github-linguist/linguist/pull/7021:

> NOTE: Search results will not reflect changes and new lanuguages until the upstream internal library is updated to pull in the changes from this release and that then deployed to GitHub's Search.

**Reply by pyhoon** [reply]
Thanks. I hope this get fix in the future.

**Reply by DecimalTurn** [reply]
Go-Enry had a [release ](https://github.com/go-enry/go-enry/releases/tag/v2.9.0) a few days ago, so as soon as the GitHub team in charge of Search makes the update, searches will include B4X.

**Reply by DecimalTurn** [reply]
@lildude It seems like the GitHub Search team has not performed the update for Go-Enry and it's been more than a month since the release. Is there a way to let them know about the update?

**Reply by lildude** [reply]
Our search team is aware there has been a new Linguist release however they no longer use go-enry so need to make time anround other priorities to update the internal library they now use.

**Reply by DecimalTurn** [reply]
Woah, ok, I guess we'll have to edit the troubleshooting section at some point to include that info.

**Reply by DecimalTurn** [reply]
Looking at GitHub Search results and all the disparities listed [here](https://github.com/go-enry/go-enry/issues/191), it seems that the library used for GitHub Search was not updated since [v7.26](https://github.com/github-linguist/linguist/releases/tag/v7.26.0) which was almost 2 years ago. 

@jorendorff, are there any intentions to update the list of languages supported by Search in the new internal library you mentioned [here](https://github.com/github-linguist/linguist/pull/6417)? I'm assuming you have to do some pre-processing of the `heuristics.yml` and `languages.yml` files, but since we've simplified a lot of regexes since v7.26, that shouldn't be too complicated, I imagine.

**Reply by DecimalTurn** [reply]
Nice to see that search for B4X works now:
https://github.com/search?q=language%3AB4X&type=code

**Comment by waruqi** [comment]
It seems that half a year has passed, and it still shows "unknown language". Besides, I have also encountered the same problem. The new languages in other Linguist 9.0.0 haven't been updated either.
